### PR TITLE
Levan explore mirror

### PR DESCRIPTION
**Pull request by crangelsmith**
Adds CineMA (Cardiac image Network for Masked Autoencoding) pretraining pipeline alongside existing ViT pretraining, with Hydra configuration management for both workflows.

## Core Changes

- **CineMA Integration**: Added `heartfm/cinema_scripts/cinema_pretrain.py` implementing masked autoencoder pretraining on cardiac MRI using the CineMA architecture from UKB dataset
- **Hydra Configuration**: Migrated ViT pretraining to `vit_pretrain_hydra.py` with structured YAML configs (`cinema_pretrain.yaml`, `vit_pretrain_config.yaml`, `output_dirs.yaml`)
- **HPC Support**: Batch scripts for single-node execution on HPC clusters for both CineMA and ViT workflows
- **Enhanced Logging**: TensorBoard integration, improved checkpoint saving with learning rate scheduler, and test set evaluation metrics

## Dependencies

Added: `hydra-core`, `omegaconf`, `safetensors`, `timm`, `einops`, `tensorboard`, `CineMA` (from GitHub)

Relaxed Python requirement: `>=3.12` → `>=3.11`

## Usage

```bash
# CineMA pretraining
python heartfm/cinema_scripts/cinema_pretrain.py data.root=/path/to/mnm train.n_epochs=50

# ViT pretraining with Hydra
python heartfm/vit_pretrain_hydra.py data.root=/path/to/mnm train.epochs=50 train.batch_size=256
```

---

💬 We'd love your input! Share your thoughts on Copilot coding agent in our [2 minute survey](https://gh.io/copilot-coding-agent-survey).

### CODE REVIEW

**Comment by crangelsmith** [comment]
@copilot can you describe this PR
